Analyticsv3Sample/AnalyticsSampleProject/StandardGoogleAuthSample.cs

```diff
@@ -0,0 +1,239 @@
+        ///      
+        /// This file can be obtained from Google Developers console:   https://console.developers.google.com/project?authuser=0
+        /// </summary>
+        /// <param name="clientSecretJson">Path to the client secret json file from Google Developers console.</param>
+        /// <param name="userName">The user to authorize.</param>
+        /// <returns>a valid AnalyticsService</returns>
+        public static AnalyticsService AuthenticateOauth(string clientSecretJson, string userName)
+        {
+            if (string.IsNullOrEmpty(userName))
+                throw new Exception("userName is required for datastore.");
+
+            if (!File.Exists(clientSecretJson))
+                throw new Exception("Cant find Client Secret Json file.");
+			
+            string[] scopes = new string[] { AnalyticsService.Scope.Analytics,                       // View and manage your Google Analytics data
+                                             AnalyticsService.Scope.AnalyticsEdit,                   // Edit Google Analytics management entities
+                                             AnalyticsService.Scope.AnalyticsManageUsers,            // Manage Google Analytics Account users by email address
+                                             AnalyticsService.Scope.AnalyticsManageUsersReadonly,    // View Google Analytics user permissions
+                                             AnalyticsService.Scope.AnalyticsProvision,              // Create a new Google Analytics account along with its default property and view
+                                             AnalyticsService.Scope.AnalyticsReadonly};              // View your Google Analytics data
+            try
+            {
+                UserCredential credential;
+                using (var stream = new FileStream(clientSecretJson, FileMode.Open, FileAccess.Read))
+                {
+                    string credPath = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal);
+                    credPath = Path.Combine(credPath, ".credentials/Analytics");
+
+                    credential = GoogleWebAuthorizationBroker.AuthorizeAsync(
+                        GoogleClientSecrets.Load(stream).Secrets,
+                        scopes,
+                        userName,
+                        CancellationToken.None,
+                        new FileDataStore(credPath, true)).Result;
+                    Console.WriteLine("Credential file saved to: " + credPath);
+                }
+
+                // Create Analytics API service.
+                var service = new AnalyticsService(new BaseClientService.Initializer()
+                    {
+                        HttpClientInitializer = credential,
+                        ApplicationName = "Analytics Authentication Sample",
+                    });
+                return service;
```

Comment:
You can just return the service without have a variable for it, I would change the code to:
return new AnalyticsService(new BaseClientService.Initializer
                    {
                        HttpClientInitializer = credential,
                        ApplicationName = "Analytics Authentication Sample",
                    });